File: trunk/bitstring/test_bitstring.c
#include "bitstring.h"
#include <string.h>
#include <stdio.h>

int main(){
	
	bitstring_lib_init();

	/*INT TEST, BYTE/BIT ORDERING*/
	fprintf(stdout, "INT TEST, BYTE/BIT ORDERING\n");

	pbitstring_t pbs1;
	pbitstring_t pbs2;
	
	for(int i=20; i<25; i++){
		pbs1 = bitstring_new_uint(0x12345, i, ORD_NONE, ORD_BE);
		bitstring_print(stdout, pbs1);
		bitstring_del(pbs1);
	}


	for(int i=20; i<25; i++){
		pbs1 = bitstring_new_uint(0x12345, i, ORD_NONE, ORD_LE);
		bitstring_print(stdout, pbs1);
		bitstring_del(pbs1);
	}
	
	
	pbs1 = bitstring_new_uint(-2, 13, ORD_NONE, ORD_BE);
	bitstring_print(stdout, pbs1);
	bitstring_del(pbs1);
	
	pbs1 = bitstring_new_uint(-2, 13, ORD_NONE, ORD_LE);
	bitstring_print(stdout, pbs1);
	bitstring_del(pbs1);
	
	/*
	pbs1 = bitstring_new_uint(0x1234, SZ_SHORT, ORD_LE, ORD_BE);
	bitstring_print(stdout, pbs1);
	pbs1 = bitstring_new_uint(0x1234, SZ_SHORT, ORD_BE, ORD_BE);
	bitstring_print(stdout, pbs1);
	pbs1 = bitstring_new_uint(0x1234, SZ_SHORT, ORD_LE, ORD_LE);
	bitstring_print(stdout, pbs1);
	pbs1 = bitstring_new_uint(0x1234, SZ_SHORT, ORD_BE, ORD_LE);
	bitstring_print(stdout, pbs1);
	*/
	// /*FLOATING POINT TEST*/
	// fprintf(stdout, "FP TEST:\n");
	// pbs1 = bitstring_new_fp(3.14, 32, FP_IEEE754);
	// bitstring_print(stdout, pbs1);
	// fprintf(stdout, "%f\n", *((float*)pbs1->data));
	// pbs1 = bitstring_new_fp(3.14, 64, FP_IEEE754);
	// bitstring_print(stdout, pbs1);
	// fprintf(stdout, "%f\n", *((double*)pbs1->data));
	
	// /*APPEND AND ALLOC TEST*/
	// fprintf(stdout, "APPEND AND ALLOC TEST\n");
	// pbs2 = bitstring_new_uint(1, 13, ORD_LE, ORD_BE);
	// for(int i=0; i<100; i++){
		// bitstring_append(pbs1, pbs2);
		//bitstring_print(stdout, pbs1);
	// }
	
	// pbs1 = bitstring_new_uint(255, 13, ORD_LE, ORD_BE);
	// bitstring_print(stdout, pbs1);
	// pbs1 = bitstring_new_uint(255, 13, ORD_LE, ORD_LE);
	// pbs1->size = 32;
	// bitstring_print(stdout, pbs1);
	
	// /*APPEND BIT*/
	// pbs1 = bitstring_new_uint(255, 8, ORD_LE, ORD_LE);
	// bitstring_print(stdout, pbs1);
	// bitstring_append_bit(pbs1, 1);
	// bitstring_print(stdout, pbs1);
	// bitstring_append_bit(pbs1, 0);
	// bitstring_print(stdout, pbs1);
	// bitstring_append_bit(pbs1, 1);
	// bitstring_print(stdout, pbs1);
	// bitstring_append_bit(pbs1, 0);
	// bitstring_print(stdout, pbs1);

	
	//bitstring_del(pbs1);
	//bitstring_del(pbs2);
	return 0;
}

File: trunk/bitstring/lua/test_bitstring.lua
require("bitstring")

ORD_NONE = 2;
ORD_BE = 1;
ORD_LE = 0;

SZ_BYTE = 8
SZ_LONG  = SZ_BYTE*4
SZ_SHORT  = SZ_BYTE*2
SZ_CHAR  = SZ_BYTE*1

FP_IEEE754 = 0
FP_IBM  = 1
FP_CRAY = 2
FP_VAX  = 3

--INT TEST, BYTE/BIT ORDERING*/
print("INT TEST, BYTE/BIT ORDERING");

for i=20,32,1 do
	pbs1 = bitstring.new_uint(0x12345, i, ORD_NONE, ORD_BE);
	print(pbs1);
end

for i=20,32,1 do
	pbs1 = bitstring.new_uint(0x12345, i, ORD_NONE, ORD_LE);
	print(pbs1);
end

pbs1 = bitstring.new_uint(-2, 13, ORD_NONE, ORD_BE);
print(pbs1);

pbs1 = bitstring.new_uint(-2, 13, ORD_NONE, ORD_LE);
print(pbs1);

pbs1 = bitstring.new_uint(0x1234, SZ_SHORT, ORD_LE, ORD_BE);
print(pbs1);
pbs1 = bitstring.new_uint(0x1234, SZ_SHORT, ORD_BE, ORD_BE);
print(pbs1);
pbs1 = bitstring.new_uint(0x1234, SZ_SHORT, ORD_LE, ORD_LE);
print(pbs1);
pbs1 = bitstring.new_uint(0x1234, SZ_SHORT, ORD_BE, ORD_LE);
print(pbs1);

--FLOATING POINT TEST
print("FP TEST:");
pbs1 = bitstring.new_fp(3.14, SZ_LONG, FP_IEEE754, ORD_LE, ORD_BE);
print(pbs1);

pbs1 = bitstring.new_fp(3.14, SZ_LONG, FP_IEEE754, ORD_BE, ORD_BE);
print(pbs1);

pbs1 = bitstring.new_fp(3.14, SZ_LONG, FP_IEEE754, ORD_LE, ORD_LE);
print(pbs1);

pbs1 = bitstring.new_fp(3.14, SZ_LONG, FP_IEEE754, ORD_BE, ORD_LE);
print(pbs1);

--APPEND, CONCAT, ALLOC TEST
print("APPEND AND ALLOC TEST");
pbs1 = bitstring.new_uint(1, 13, ORD_NONE, ORD_BE);
pbs2 = bitstring.new_sint(-2, 13, ORD_NONE, ORD_BE);
for i=1,10,1 do
	print(pbs1);
	pbs1:append(pbs2);
end

--APPEND BIT
pbs1 = bitstring.new_uint(255, 8, ORD_LE, ORD_LE);
for i=1,8,1 do
	print(pbs1);
	pbs1:append_bit(0);
end

pbs1 = bitstring.new_uint(0, 8, ORD_LE, ORD_BE);
pbs2 = bitstring.new_uint(255, 8, ORD_LE, ORD_BE);
pbs3 = pbs1 .. pbs2;
print(pbs3);

